app/controllers/api/v1/admin/user_skills_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Api::V1::Admin::UserSkillsController < Api::V1::BaseController
+  before_action :authenticate_user!
+
+  def index
+  end
+
+  def create
+    @user_skill = UserSkill.new(user_skill_params)
+    authorize @user_skill
+    if @user_skill.save
+      respond_with :api, :v1, :admin, UserSkill.last
+    else
+      render json: { errors: "Something went wrong" }, status: 422
+    end
+  end
+
+  def new
+  end
+
+  def update
+  end
+
+  def destroy
+    params[:user_skill][:skill_id] = Skill.find_by(name: params[:user_skill][:name]).id
```

Comment:
this is `unsafe`. And I actually, don't understand what an why you are doing here.